src/lib.rs

```diff
@@ -208,15 +250,7 @@ impl Matcher for TreeSitterMatcher<'_> {
                 text_len: haystack.len(),
             }
         });
-        if matches_info.matches.is_empty() {
-            return Ok(None);
-        }
-        let match_ = matches_info.matches.remove(0);
```

Comment:
...and so we can't just blindly "pop + return" one match at a time, we have to keep them all around and give it back the first one that starts after the (calculated via the pseudo-invariant) file contents offset of the substring that it's passing us